### DISCUSSION TITLE
Moving to a monorepo, starting with shared UI components

### DISCUSSION THREAD

**Discussion by pkspyder007** [codu-code/codu#601] [Ideas]
**Introduction:**

Hey everyone,

Time for a cozy chat about a big change on our horizon – we're thinking of throwing our shared UI library and our app into one big happy monorepo. 🎉 

**The Problem:**

So, what's the beef? Well, we've noticed a couple of things that bug us:

1. **Inconsistency:** The dreaded inconsistency monster! We've got different teams with their own copies of UI components, and it's turning our beautiful UI into a patchwork quilt. Not exactly the look we're going for.

2. **Maintenance Madness:** Maintaining all these separate UI components in different repositories is like trying to herd cats. It's chaos! A monorepo could give us some peace and quiet.

**The Plan:**

Now, here's the juicy part – the monorepo. It's like putting all our eggs in one basket, but in a good way:

1. **One UI Library to Rule Them All:** We're thinking of having a single source for our UI components. No more "Who copied that component last? Oops, forgot to update it here." It's all in one place.

2. **Easy Peasy Maintenance:** With everything under one roof, maintaining our UI components is a piece of cake. Bug fixes, feature updates, and documentation? Do it once, and it's done everywhere.

3. **Teamwork Makes the Dream Work:** A monorepo encourages collaboration. Developers from different projects can work together on UI components, leading to some real magic.

4. **Dependencies, Simplified:** Keeping our dependencies in check becomes a walk in the park. No more dependency conflicts or stress. 🌈

**Questions and Comments:**

We're all ears for your thoughts on this grand idea! Feel free to chime in with your wisdom and worries:

1. **What's your take on jumping into a monorepo, starting with our shared UI library?**

2. **Any particular pros and cons you see with this move? Let's hear 'em! 😄**

3. **Got ideas on how to make this transition smooth and sweet for the whole gang?**

4. **Tools, tips, and tricks – what should we consider for our one-app-to-rule-them-all monorepo setup?**

Hit us up with your insights, stories, and questions. Your input means the world to us!

Cheers,
@pkspyder007

**Comment by DarrachBarneveld** [comment]
This is a fantastic idea. Reusable layout/styled components is a must. 

Surely the switch wont be too bad considering the tools currently used already? I was wondering why thier wasn't at least a UI folder in the components directory. 

Would love to help in refactoring :)
